src/lucky/html_builder.cr

```diff
@@ -45,8 +45,9 @@ module Lucky::HTMLBuilder
 
   macro generate_needy_initializer
     {% if !@type.abstract? %}
+      {% sorted_assigns = ASSIGNS.sort_by { |dec| dec.type.resolve.nilable? || dec.value ? 1 : 0 } %}
```

Comment:
I think something like this may work and ensure that false and nil defaults work too:

decl.default || decl.default == nil == decl.default == false

I’m not sure but that’s my hunch

Reply:
If the type is `thing : String?`, then there's no value at all. It doesn't seem to even equate to `nil` from what I can tell. Without the resolve, it won't compile. But I'll add in the check for it having a value of nil and false.